tests/run-pass/global_allocator.rs

```diff
@@ -0,0 +1,29 @@
+use std::alloc::{GlobalAlloc, Layout, System};
+
+#[global_allocator]
+static ALLOCATOR: Allocator = Allocator;
+
+struct Allocator;
+
+unsafe impl GlobalAlloc for Allocator {
+    unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+        // use specific size to avoid getting triggered by rt
+        if layout.size() == 123 {
+            println!("Allocated!")
+        }
+
+        System.alloc(layout)
+    }
+
+    unsafe fn dealloc(&self, ptr: *mut u8, layout: Layout) {
+        if layout.size() == 123 {
+            println!("Dellocated!")
+        }
+
+        System.dealloc(ptr, layout)
+    }
+}
+
+fn main() {
+    Box::new([0_u8; 123]);
```

Comment:
Could you extend the test with something like this, which can only work when using custom allocators?

```rust
    // Only okay because we explicitly set a global allocator that uses the system allocator!
    let l = Layout::from_size_align(1, 1).unwrap();
    let ptr = Global.allocate(l).unwrap().as_non_null_ptr(); // allocating with Global...
    unsafe { System.deallocate(ptr, l); } /// ... and deallocating with System.
```

Reply:
Alright, I've modified the test to use that instead of printing on a specific allocation size.

Reply:
I liked the printing part, the intention was to have both. :)

Reply:
Alright, I've readded the printing and tested the reverse: allocating with system and deallocating with global.